thrust/thrust/iterator/transform_input_output_iterator.h

```diff
@@ -30,101 +29,168 @@
 #elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
 #  pragma system_header
 #endif // no system header
-#include <thrust/iterator/detail/transform_input_output_iterator.inl>
+
+#include <thrust/detail/type_traits.h>
```

Comment:
Are we using that for anything really? I only see `invoke_result_t`. That could come from cuda::std::